terraform/environment/region/health_checks.tf

```diff
@@ -0,0 +1,142 @@
+resource "aws_sns_topic" "service_health_checks_global" {
+  name                                     = "${data.aws_default_tags.current.tags.environment-name}-service-health-checks"
+  kms_master_key_id                        = data.aws_kms_alias.sns_kms_key_alias_global.target_key_id
+  application_failure_feedback_role_arn    = data.aws_iam_role.sns_failure_feedback.arn
+  application_success_feedback_role_arn    = data.aws_iam_role.sns_success_feedback.arn
+  application_success_feedback_sample_rate = 100
+  firehose_failure_feedback_role_arn       = data.aws_iam_role.sns_failure_feedback.arn
+  firehose_success_feedback_role_arn       = data.aws_iam_role.sns_success_feedback.arn
+  firehose_success_feedback_sample_rate    = 100
+  http_failure_feedback_role_arn           = data.aws_iam_role.sns_failure_feedback.arn
+  http_success_feedback_role_arn           = data.aws_iam_role.sns_success_feedback.arn
+  http_success_feedback_sample_rate        = 100
+  lambda_failure_feedback_role_arn         = data.aws_iam_role.sns_failure_feedback.arn
+  lambda_success_feedback_role_arn         = data.aws_iam_role.sns_success_feedback.arn
+  lambda_success_feedback_sample_rate      = 100
+  sqs_failure_feedback_role_arn            = data.aws_iam_role.sns_failure_feedback.arn
+  sqs_success_feedback_role_arn            = data.aws_iam_role.sns_success_feedback.arn
+  sqs_success_feedback_sample_rate         = 100
+  provider                                 = aws.global
+}
+
+resource "aws_route53_health_check" "service_health_check" {
+  fqdn              = aws_route53_record.app.fqdn
+  reference_name    = "${substr(data.aws_default_tags.current.tags.environment-name, 0, 20)}-service-hc"
+  port              = 443
+  type              = "HTTPS"
+  failure_threshold = 1
```

Comment:
Was there a reason for making this an instant fail? Could we maybe increase the threshold and decrease the request interval to account for blips (unless we care about blips (such a technical term 😆)?)

Reply:
I don't know what the pattern for dependency failures would look like, so starting here and we can adjust as we learn more.